### PR TITLE
Close the client if there is an error in startup

### LINKED ISSUE #1282: If Timelock Server crashes owing to an exception, the Atomix client is not closed.
Steps to reproduce: 
1. Add a line at the end of the `run()` method in `TimeLockServer` that throws an exception (or error, if you prefer).
2. Run timelock server (assuming default settings), watch it crash
3. Curl `localhost:8700`. 
Expected: connection refused; actual: empty reply from server (so we're still listening)

### PR DESCRIPTION
**Pull request by hsaraogi**
fixes #1282.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1314)

### CODE REVIEW

**Comment by SerialVelocity** [comment]
This probably won't work. See #1282 for details.

File: atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java [hunk 1]
```diff
@@ -106,24 +101,32 @@ private void setup(AtomixReplica localNode, TimeLockServerConfiguration configur
...
+                localMember,
+                leaderInfo);
+
+        environment.jersey().register(HttpRemotingJerseyFeature.DEFAULT);
+        environment.jersey().register(new TimeLockResource(clientToServices));
     }
 
-    private Map<String, TimeLockServices> createTimeLockServicesForClients(
+    private static Map<String, TimeLockServices> createTimeLockServicesForClients(
             AtomixReplica localNode,
```
Comment (hsaraogi) [inline atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java]: ``replica``

File: atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    @After
+    public void sendShutdownToListeners() {
+        LifeCycle event = mock(LifeCycle.class);
+        listeners.forEach(listener -> listener.lifeCycleStopping(event));
+        listeners.forEach(listener -> listener.lifeCycleStopped(event));
+    }
+
+    @Test
+    public void atomixIsRunningAfter() throws IOException {
```
Comment (jeremyk-91) [inline atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java]: nit: running after successful startup?

File: atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+public class TimeLockServerTest {
+    private static final TimeLockServerConfiguration TIMELOCK_CONFIG = new TimeLockServerConfiguration(
+            ImmutableAtomixConfiguration.builder()
+                    .storageLevel(StorageLevel.MEMORY)
+                    .build(),
+            ImmutableClusterConfiguration.builder()
+                    .localServer(new Address("localhost:12345"))
+                    .addServers(new Address("localhost:12345"))
+                    .build(),
+            ImmutableSet.of("localhost:12345"));
```
Comment (jeremyk-91) [inline atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java]: nit: we probably want one or more constants for the repeated `"localhost:12345"`.

**Pull request merged**